static/js/server_events_dispatch.js

```diff
@@ -356,6 +356,8 @@ exports.dispatch_normal_event = function dispatch_normal_event(event) {
 
     case 'delete_message':
         var msg_id = event.message_id;
+        var message = message_store.get(msg_id);
+        unread_ops.mark_message_as_read(message);
```

Comment:
Does this do the right thing in the case that the message had already been read?  (I can imagine trying to double-mark something as read throwing an exception).

Reply:
Yes, sorry should have mentioned in the PR itself. `mark_messages_as_read` passes the message list to  `unread.get_unread_messages`, which returns all the unread messages out of the given list. So double marking something as read would not occur

Reply:
Cool.  I''ll merge a quick PR adding a comment to that effect, just to make things more readable.